src/nlp/features.py

```diff
@@ -400,10 +400,17 @@ def encode_nested_example(schema, obj):
     elif isinstance(schema, Sequence):
         # We allow to reverse list of dict => dict of list for compatiblity with tfds
         if isinstance(schema.feature, dict):
-            return dict(
-                (k, [encode_nested_example(sub_schema, o) for o in sub_obj])
-                for k, (sub_schema, sub_obj) in utils.zip_dict(schema.feature, obj)
-            )
+            if isinstance(obj, (list, tuple)):
+                zipped_dict = utils.zip_dict(schema.feature, *obj)
+            else:
+                zipped_dict = utils.zip_dict(schema.feature, obj)
+            list_dict = {}
+            for k, dict_tuples in zipped_dict:
+                sub_schema = dict_tuples[0]
+                sub_objs = dict_tuples[1:] if len(dict_tuples) > 2 else dict_tuples[1]
```

Comment:
need this if statement here to be sure all previous datasets work exactly the same as before

Reply:
Ok but won't this break if `obj` is a list (ie. the special case of qa4mre) but with only one element (i.e. probably `len(dict_tuples)`== 2)?
Maybe the test here should be again `if isinstance(obj, (list, tuple))`?

Reply:
True! There was currently no case of this, but could be - I think the logic as it is now, is better.